public/app/features/alerting/AlertTabCtrl.ts

```diff
@@ -272,20 +274,34 @@ export class AlertTabCtrl {
           foundTarget = firstTarget;
         } else {
           this.error = 'Could not find any metric queries';
+          return;
         }
       }
 
       const datasourceName = foundTarget.datasource || this.panel.datasource;
-      this.datasourceSrv.get(datasourceName).then(ds => {
-        if (!ds.meta.alerting) {
-          this.error = 'The datasource does not support alerting queries';
-        } else if (ds.targetContainsTemplate && ds.targetContainsTemplate(foundTarget)) {
-          this.error = 'Template variables are not supported in alert queries';
-        } else {
-          this.error = '';
-        }
-      });
+      promises.push(
+        this.datasourceSrv.get(datasourceName).then(
+          (foundTarget => (ds: DataSourceApi) => {
+            if (!ds.meta.alerting) {
+              return this.$q.reject('The datasource does not support alerting queries');
+            } else if (ds.targetContainsTemplate && ds.targetContainsTemplate(foundTarget)) {
+              return this.$q.reject('Template variables are not supported in alert queries');
+            }
+            return this.$q.resolve();
+          })(foundTarget)
+        )
+      );
     }
+    this.$q.all(promises).then(
+      () => {
+        this.error = '';
+        this.$scope.$apply();
+      },
+      e => {
+        this.error = e;
+        this.$scope.$apply();
+      }
+    );
```

Comment:
think this could be improved (refactored) further by extract function, and using async await, try catch. But this is fine for now.

Reply:
I think for loop with await is not a good practice